components/inputs/test/input-time-range.test.js

```diff
@@ -47,10 +47,84 @@ describe('d2l-input-time-range', () => {
 			expect(elem.endValue).to.equal('23:00:00');
 		});
 
-		it('should default start and end values to undefined', async() => {
+		it('should default start and end values to 12 am and 12:30 am', async() => {
 			const elem = await fixture(basicFixture);
-			expect(elem.startValue).to.equal(undefined);
-			expect(elem.endValue).to.equal(undefined);
+			expect(elem.startValue).to.equal('00:00:00');
+			expect(elem.endValue).to.equal('00:30:00');
+		});
+
+		describe('validation', () => {
+			it('should be valid if start time and no end time and start time changed to value before default end', async() => {
+				const elem = await fixture(basicFixture);
+				const inputElem = getChildElem(elem, 'd2l-input-time.d2l-input-time-range-start');
+				inputElem.value = '00:15:00';
+				setTimeout(() => dispatchEvent(inputElem, 'change'));
+				await oneEvent(elem, 'change');
+				expect(elem.startValue).to.equal('0:15:00');
+				expect(elem.endValue).to.equal('00:30:00');
+				expect(elem.invalid).to.be.false;
+				expect(elem.validationError).to.be.null;
+			});
+
+			it('should be valid if end time and no start time', async() => {
```

Comment:
Just to confirm... it's not that there's no start value in the range, it's that one wasn't provided and we default it to midnight? Like visually to the user there's no way for a range to be missing a start and/or end value -- it'll always have both?

Reply:
Yup that's correct - I copy/pasted from the date range tests and forgot to update the context here.